### PR TITLE
v1.13 Backports 2024-03-11

### PR DESCRIPTION
**Pull request by jibi**
* [x] #30988 (@tommyp1ckles)
 * [x] #31014 (@tommyp1ckles)
 * [x] #31159 (@pchaigno)
 * [x] #31196 (@aanm)
 * [x] #31015 (@learnitall)

PRs skipped due to conflicts:

 * #31292 (@chaunceyjiang) cc @sayboras 

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 30988 31014 31159 31196 31015
```

### CODE REVIEW

**Comment by jibi** [comment]
/test-backport-1.13

File: .github/actions/unit-tests/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```
Comment (aanm) [inline .github/actions/unit-tests/prepare.sh]: @jibi my commit is a little bit different than the changes being backported :confused:

**Review by aanm — CHANGES_REQUESTED** [review]

File: .github/actions/unit-tests/prepare.sh (hunk 1 above)
Comment (jibi) [reply .github/actions/unit-tests/prepare.sh]: this file should have been removed altogether, fixed now

**Comment by sayboras** [comment]
>  * #31292 (@chaunceyjiang) cc @sayboras 

Noted and thanks, I will take care of the backport manually for 1.13.

File: .github/workflows/integration-test.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        description: "SHA under test (head of the PR branch)."
+        required: true
+      extra-args:
+        description: "[JSON object] Arbitrary arguments passed from the trigger comment via regex capture group. Parse with 'fromJson(inputs.extra-args).argName' in workflow."
+        required: false
+        default: '{}'
+  push:
+    branches:
+    - v1.14
+    - ft/v1.14/**
```
Comment (aanm) [inline .github/workflows/integration-test.yaml]: v1.13

**Comment by jibi** [comment]
/test-backport-1.13

**Review by aanm — APPROVED** [review]

**Comment by jibi** [comment]
/test-backport-1.13

**Review by dylandreimerink — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
My PR looks good. Thanks Jibi!

**Review by learnitall — APPROVED** [review]

**Review by learnitall — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Pull request merged**